Found an issues with Functions.T4 when genericArg was null

burrowj opened the pull request:
When genericArg was null there were errors.  
Also when return type is int then we need the FirstorDefault() when query is SqlQueryRawAsync

R4ND3LL commented:
i'll do some testing on this on monday and merge provided the tests pass.

Thanks!

R4ND3LL commented:
your change is in build 1.3.6

burrowj commented:
MarkI’ve found that the return grows generally for table functions in sql aren’t being created properly.  I’d like to debug the code but don’t know how to debug the “dot net “ command line.  I’ve setup a test project under the “tests” folder and can run tests on code tests but how do I debug the command lines?Sent from my iPhoneOn 20 Dec 2023, at 5:12 am, Mark Randell ***@***.***> wrote:﻿
I merged your pull request BUT had to roll back your FirstOrDefault addition because it broke the code.  if you have a special function that only  works with FirstOrDefault, provide a sample to me and I will add support for it and ensure it does not break other scenarios

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you authored the thread.Message ID: ***@***.***>

R4ND3LL commented:
I have tests with table functions.  Provide a sample of one that is not working for you and I'll get it fixed up.

burrowj commented:
Table return function like this

I have removed the inner code but you get the drift



CREATE FUNCTION [dbo].[fnt_RaceGroupEntryCategoryAncestorHierarchy]

(

       @EntityId uniqueidentifier

)

RETURNS @T table(id uniqueidentifier, Name nvarchar(80), ParentEntityId
uniqueidentifier, IsEntrantCategory bit, IsRaceDivisionCategory bit,
IsResultsCategory bit, Level int )

AS

BEGIN

END

GO





*From:* Mark Randell ***@***.***>
*Sent:* Saturday, December 23, 2023 12:50 AM
*To:* R4ND3LL/EntityFrameworkRuler ***@***.***>
*Cc:* John ***@***.***>; Author <
***@***.***>
*Subject:* Re: [R4ND3LL/EntityFrameworkRuler] Found an issues with
Functions.T4 when genericArg was null (PR #31)



I have tests with table functions. Provide a sample of one that is not
working for you and I'll get it fixed up.

—
Reply to this email directly, view it on GitHub
<https://github.com/R4ND3LL/EntityFrameworkRuler/pull/31#issuecomment-1867710686>,
or unsubscribe
<https://github.com/notifications/unsubscribe-auth/AESHI4KIC4JKWXG477E2TYDYKWFYZAVCNFSM6AAAAABAVWWFROVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTQNRXG4YTANRYGY>
.
You are receiving this because you authored the thread.[image: Image
removed by sender.]Message ID: <R4ND3LL/EntityFrameworkRuler/pull/31/
***@***.***>
